ldap/ldap.go

```diff
@@ -0,0 +1,96 @@
+package ldap
+
+import (
+	"crypto/tls"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ldap "gopkg.in/ldap.v2"
+)
+
+// LDAPAuthenticator provides authentication and authorization of users against an LDAP service.
+type LDAPAuthenticator struct {
+	url  string
+	port string
+	path string
+	conn *ldap.Conn
+}
+
+// NewLDAPAuthenticator constructs an LDAPAuthenticator. It requires a url and port to the LDAP
+// server. It also requires a path to user resources that can be passed to an LDAP query.
+func NewLDAPAuthenticator(url, port, path string) (*LDAPAuthenticator, error) {
```

Comment:
call this something like user service, (ldap is in the package name) and authenticator sounds like an interface but isn't the one from gimlet. 

The constructor doesn't do much, but I think it's probably ok keep it like this (vs a validate method and literal constructor syntax)